server/requestorserver/server.go

```diff
@@ -159,11 +159,22 @@ var corsOptions = cors.Options{
 	AllowedMethods: []string{http.MethodGet, http.MethodPost, http.MethodDelete},
 }
 
+func (s *Server) createRouters() (rootRouter, apiRouter *chi.Mux) {
+	rootRouter = chi.NewRouter()
+	rootRouter.Route(s.conf.ApiPrefix, func(r chi.Router) {
+		// r will be a *chi.Mux, see https://github.com/go-chi/chi/blob/c9e87efe9691a63d6a89de8bbd16b04fe4d6640e/mux.go#L262
+		// We don't use a chi.Router, because we need the {NotFound,MethodNotAllowed}Handlers methods of *chi.Mux later on that
+		// are not part of the chi.Router interface.
+		apiRouter = r.(*chi.Mux)
```

Comment:
I think we can avoid this type assertion, as wel as simplify the implementation of this feature as follows. Based on the original code:
* Write a new unexported method on `Server` that, given a `router`, either just returns it if `s.conf.ApiPrefix` is empty; or it creates and returns a new `rootRouter` to which the `router` is attached using `rootRouter.Mount(s.conf.ApiPrefix, router.Handler())`;
* Return the return value of that method at the end of `Server.ClientHandler()` and `Server.Handler()`.

Reply:
Thanks, that's much better;  I've followed your suggestion.

Reply:
Thanks, this is better. I noticed you mount `router.Handler()` unconditionally to a new router. In most cases `ApiPrefix` will be empty, but this will cause a (small) performance penalty anyway, right? Would it not be better to just return the router in the new method in this case?